pkg/controller/nodes/task/handler_test.go

```diff
@@ -150,41 +151,59 @@ func Test_task_Setup(t *testing.T) {
 		defaultPluginID string
 	}
 	tests := []struct {
-		name     string
-		registry PluginRegistryIface
-		fields   wantFields
-		wantErr  bool
+		name                 string
+		registry             PluginRegistryIface
+		enabledPluginsConfig map[string]config.EnabledPlugins
+		fields               wantFields
+		wantErr              bool
 	}{
-		{"no-plugins", testPluginRegistry{}, wantFields{}, false},
+		{"no-plugins", testPluginRegistry{}, map[string]config.EnabledPlugins{}, wantFields{}, false},
 		{"no-default-only-core", testPluginRegistry{
 			core: []pluginCore.PluginEntry{corePluginEntry}, k8s: []pluginK8s.PluginEntry{},
+		}, map[string]config.EnabledPlugins{
+			corePluginType: {DefaultPluginTasks: []string{corePluginType}},
 		}, wantFields{
 			pluginIDs: map[pluginCore.TaskType]string{corePluginType: corePluginType},
 		}, false},
 		{"no-default-only-k8s", testPluginRegistry{
 			core: []pluginCore.PluginEntry{}, k8s: []pluginK8s.PluginEntry{k8sPluginEntry},
+		}, map[string]config.EnabledPlugins{
+			k8sPluginType: {DefaultPluginTasks: []string{k8sPluginType}},
 		}, wantFields{
 			pluginIDs: map[pluginCore.TaskType]string{k8sPluginType: k8sPluginType},
 		}, false},
-		{"no-default", testPluginRegistry{
-			core: []pluginCore.PluginEntry{corePluginEntry}, k8s: []pluginK8s.PluginEntry{k8sPluginEntry},
+		{"no-default", testPluginRegistry{}, map[string]config.EnabledPlugins{
+			corePluginType: {DefaultPluginTasks: []string{corePluginType}},
+			k8sPluginType:  {DefaultPluginTasks: []string{k8sPluginType}},
 		}, wantFields{
-			pluginIDs: map[pluginCore.TaskType]string{corePluginType: corePluginType, k8sPluginType: k8sPluginType},
+			pluginIDs: map[pluginCore.TaskType]string{},
```

Comment:
@EngHabu can you let me know if this change is correct? we shouldn't expect any plugins right if the registry is empty?

Reply:
👍